src/message.rs

```diff
@@ -570,6 +570,20 @@ impl Message {
         .await
     }
 
+    pub async fn get_sender_name(&self, context: &Context) -> String {
+        if let Some(name) = self.param.get(Param::OverrideDisplayname) {
+            name.to_string()
+        } else {
+            match Contact::load_from_db(context, self.from_id).await {
```

Comment:
We are calling Contact::load_from_db() here every time this fn is called, which may be often if the UI loads all messages instead of just the ones shown to the user. Before the UI just called `contact.get_display_name()`.
- How bad is this?

Possible solutions:
- Do something like `dc_msg_get_summary()` does: In the FFI, add an optional param `chat`, when it is given, no SQL is needed. In the Android bindings this resulted in code handling with raw C data (that's what `dc_msg_get_summary` does) and I finally decided I should not do this.
- This can't be done by a single SQL query as both the display name and the address may be needed, but of course I could write a fn that only loads these two from SQL. We would still have an SQL query where we had none before.
- MY CURRENT FAVORITE: Rename it to get_override_sender_name() and return only `self.param.get(Param::OverrideDisplayname)`. If it is None/NULL, the UI has to call contact.get_display_name() itself.

Reply:
there is a function get_override_sender_name() now, so, i assume, this can be marked as resolved.